REGRESSION(288458@main): <input type=checkbox switch> no longer calls updateTouchEventHandler()

annevk opened the pull request:
#### 3138a5facd50b7419690bba1267edb85a75c3f34
<pre>
REGRESSION(288458@main): &lt;input type=checkbox switch&gt; no longer calls updateTouchEventHandler()
<a href="https://bugs.webkit.org/show_bug.cgi?id=287181">https://bugs.webkit.org/show_bug.cgi?id=287181</a>
<a href="https://rdar.apple.com/130619586">rdar://130619586</a>

Reviewed by Aditya Keerthi.

When the parser calls initializeInputTypeAfterParsingOrCloning() the
switch attribute is not initialized yet and we have not updated
initializeInputTypeAfterParsingOrCloning() to account for the switch
attribute, so when descendants of that method call
updateTouchEventHandler(), isSwitch() still returns false. This means
no touch event listeners are registered. (The switch attribute is
initialized immediately after
initializeInputTypeAfterParsingOrCloning() is invoked.)

So we update the comment in HTMLInputElement::attributeChanged to
indicate we solely rely on updateUserAgentShadowTree() for the switch
attribute and move up updateTouchEventHandler() before the early return
to ensure it&apos;s always called when the switch attribute is initialized.

Then we clean up TestExpectations. We skip all the
&lt;input type=checkbox switch&gt; tests that depend on ENABLE(TOUCH_EVENTS).

And finally we update pointer-tracking-there-and-back-again[-rtl].html.
We port the changes 285662@main made to -rtl to non-rtl, but we remove
the waitUntilDone() infrastructure as we use reftest-wait. We add a
wait(0.21) call to account for 278311@main. We reduce the duration of
the touch moves, but not to 0 as that makes them not register. In -rtl
we also adjust the move to the left so it overshoots x by 1 pixel. This
makes it match the non-rtl version better. And we explicitly log the
current switched state so we can instantly tell if there&apos;s some kind of
regression on that front.

* LayoutTests/fast/forms/switch/click-disabled.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-expected.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-rtl-expected.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-rtl.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again.html:
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::attributeChanged):

Canonical link: <a href="https://commits.webkit.org/290692@main">https://commits.webkit.org/290692@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fd48e5c4b0c43d637eb938cee34f6ca460be58ba

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90631 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10163 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45571 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95643 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41413 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92685 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10561 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18482 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69754 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27305 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93632 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8070 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82199 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50095 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7821 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/36584 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40543 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78140 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37643 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97472 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17823 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/13109 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78778 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18081 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78019 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77970 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22412 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/21036 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11145 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14313 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17833 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23179 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17572 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21028 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19356 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again.html:
```diff
@@ -12,20 +13,22 @@
+    await window.eventSender.mouseDown();
+    await window.eventSender.mouseMoveTo(x + width, y);
+    await window.eventSender.mouseMoveTo(x, y);
+    await window.eventSender.mouseUp();
```
The synchronous version of these eventSender functions isn't awaitable.
```suggestion
    await window.eventSender.asyncMouseMoveTo(x, y);
    await window.eventSender.asyncMouseDown();
    await window.eventSender.asyncMouseMoveTo(x + width, y);
    await window.eventSender.asyncMouseMoveTo(x, y);
    await window.eventSender.asyncMouseUp();
```

pxlcoder replied:
Do we even need to use the async versions?

charliewolfe replied:
If the test passes with them there's no reason not to. I've been trying to adopt them in tests so that they behave correctly with the `--site-isolation` option.

annevk committed:
REGRESSION(288458@main): <input type=checkbox switch> no longer calls updateTouchEventHandler()
https://bugs.webkit.org/show_bug.cgi?id=287181
rdar://130619586

Reviewed by Aditya Keerthi.

When the parser calls initializeInputTypeAfterParsingOrCloning() the
switch attribute is not initialized yet and we have not updated
initializeInputTypeAfterParsingOrCloning() to account for the switch
attribute, so when descendants of that method call
updateTouchEventHandler(), isSwitch() still returns false. This means
no touch event listeners are registered. (The switch attribute is
initialized immediately after
initializeInputTypeAfterParsingOrCloning() is invoked.)

So we update the comment in HTMLInputElement::attributeChanged to
indicate we solely rely on updateUserAgentShadowTree() for the switch
attribute and move up updateTouchEventHandler() before the early return
to ensure it's always called when the switch attribute is initialized.

Then we clean up TestExpectations. We skip all the
<input type=checkbox switch> tests that depend on ENABLE(TOUCH_EVENTS).

And finally we update pointer-tracking-there-and-back-again[-rtl].html.
We port the changes 285662@main made to -rtl to non-rtl, but we remove
the waitUntilDone() infrastructure as we use reftest-wait. We add a
wait(0.21) call to account for 278311@main. We reduce the duration of
the touch moves, but not to 0 as that makes them not register. In -rtl
we also adjust the move to the left so it overshoots x by 1 pixel. This
makes it match the non-rtl version better. And we explicitly log the
current switched state so we can instantly tell if there's some kind of
regression on that front.

* LayoutTests/fast/forms/switch/click-disabled.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-expected.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-rtl-expected.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again-rtl.html:
* LayoutTests/fast/forms/switch/pointer-tracking-there-and-back-again.html:
* LayoutTests/platform/ios/TestExpectations:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::attributeChanged):

Canonical link: https://commits.webkit.org/290692@main
